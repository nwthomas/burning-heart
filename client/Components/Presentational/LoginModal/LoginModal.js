import React, { useContext } from "react";
import {
  Modal,
  View,
  Text,
  StyleSheet,
  Dimensions,
  TouchableHighlight,
  ActivityIndicator
} from "react-native";
import { Store } from "../../store/store";
import { closeModal } from "../../store/actions";

const { width, height } = Dimensions.get("window"); // Get window dimensions

const SignUpModal = props => {
  const { state, dispatch } = useContext(Store);
  const { loginStart, loginSuccess, loginError, message } = state;
  const goToLogin = e => {
    e.preventDefault();
    if (loginSuccess) props.history.push("/");
    closeModal(dispatch);
  };
  return (
    <Modal
      animationType="slide"
      transparent={false}
      visible={loginStart || loginSuccess || loginError}
    >
      <View style={styles.modal}>
        <Text style={styles.modalText}>
          {message ? message : "Logging in..."}
        </Text>
        <TouchableHighlight
          underlayColor={"#0E30F050"} // Last two numbers indicate opacity of color
          onPress={goToLogin}
          style={styles.submitBtn}
        >
          {loginStart ? (
            <ActivityIndicator size="small" color="#ffffff" />
          ) : (
            <Text style={styles.btnText}>Okay</Text>
          )}
        </TouchableHighlight>
      </View>
    </Modal>
  );
};

const styles = StyleSheet.create({
  modal: {
    flex: 1,
    width: width,
    height: height,
    zIndex: 10,
    backgroundColor: "#ffffff",
    justifyContent: "center",
    alignContent: "center"
  },
  modalText: {
    alignSelf: "center",
    fontFamily: "Roboto-Medium",
    fontSize: 20,
    textAlign: "center",
    marginLeft: 40,
    marginRight: 40,
    marginBottom: 20
  },
  submitBtn: {
    alignSelf: "stretch",
    justifyContent: "center",
    marginLeft: 40,
    marginRight: 40,
    height: 45,
    borderWidth: 1,
    borderRadius: 5,
    borderColor: "#0E30F0",
    backgroundColor: "#0E30F0",
    marginBottom: 20,
    marginTop: 10
  },
  btnText: {
    color: "#ffffff",
    alignSelf: "center",
    fontFamily: "Roboto-Medium",
    fontSize: 16
  }
});

export default SignUpModal;
