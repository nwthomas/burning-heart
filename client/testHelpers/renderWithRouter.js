import React from "react";
import { NativeRouter as Router } from "react-router-native";
import { render } from "react-testing-library";
import { createMemoryHistory } from "history";

const renderWithRouter = (
  ui,
  {
    route = "/",
    history = createMemoryHistory({ initialEntries: [route] })
  } = {}
) => {
  return {
    ...render(<Router history={history}>{ui}</Router>),
    history
  };
};

export default renderWithRouter;
