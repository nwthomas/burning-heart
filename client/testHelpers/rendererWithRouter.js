import React from "react";
import { NativeRouter as Router } from "react-router-native";
import renderer from "react-test-renderer";
import { createMemoryHistory } from "history";

const rendererWithRouter = (
  ui,
  {
    route = "/",
    history = createMemoryHistory({ initialEntries: [route] })
  } = {}
) => {
  return {
    ...renderer.create(<Router history={history}>{ui}</Router>),
    history
  };
};

export default rendererWithRouter;
