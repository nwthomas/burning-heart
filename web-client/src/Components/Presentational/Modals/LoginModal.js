import React, { Component } from "react";
import { connect } from "react-redux";
import { closeLoginModal } from "../../../store/actions";

import LoadingAnimation from "./LoadingAnimation";

class LoginModal extends Component {
  handleButton = e => {
    e.preventDefault();
    if (this.props.loginStart) {
      return false;
    } else if (this.props.loginSuccess) {
      this.props.closeLoginModal();
      return this.props.history.push("/home");
    } else {
      this.props.closeLoginModal();
    }
  };

  componentDidMount() {
    window.scrollTo(0, 0);
    const body = document.querySelector("body");
    body.classList.toggle("stopScroll");
  }

  componentWillUnmount() {
    const body = document.querySelector("body");
    body.classList.toggle("stopScroll");
  }

  render() {
    return (
      <div className="modal__container">
        <div className="modal__box">
          <h2 className="modal__title">
            {this.props.loginStart ? "Logging In" : this.props.message}
          </h2>
          <div className="modal__buttons modal__buttons--center">
            <button className="modal__button" onClick={this.handleButton}>
              {this.props.loginStart ? <LoadingAnimation /> : "Okay"}
            </button>
          </div>
        </div>
      </div>
    );
  }
}

const mapStateToProps = state => ({
  loginStart: state.loginReducer.loginStart,
  loginSuccess: state.loginReducer.loginSuccess,
  loginError: state.loginReducer.loginError,
  modalOpen: state.loginReducer.modalOpen,
  message: state.loginReducer.message
});

const mapActionsToProps = {
  closeLoginModal
};

export default connect(
  mapStateToProps,
  mapActionsToProps
)(LoginModal);
