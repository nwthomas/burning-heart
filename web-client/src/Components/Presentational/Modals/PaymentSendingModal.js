import React from "react";
import { connect } from "react-redux";
import { closePaymentModal } from "../../../store/actions";

import LoadingAnimation from "./LoadingAnimation";

const PaymentSendingModal = props => {
  const handleButton = e => {
    e.preventDefault();
    if (props.loginStart) {
      return false;
    } else {
      props.closePaymentModal();
    }
  };
  return (
    <div className="modal__container">
      <div className="modal__box">
        <h2 className="modal__title">
          {props.makePaymentStart ? "Sending Donation" : props.message}
        </h2>
        <div className="modal__buttons modal__buttons--center">
          <button className="modal__button" onClick={handleButton}>
            {props.makePaymentStart ? <LoadingAnimation /> : "Okay"}
          </button>
        </div>
      </div>
    </div>
  );
};

const mapStateToProps = state => ({
  message: state.donorReducer.message,
  makePaymentStart: state.donorReducer.makePaymentStart
});

const mapActionsToProps = {
  closePaymentModal
};

export default connect(
  mapStateToProps,
  mapActionsToProps
)(PaymentSendingModal);
