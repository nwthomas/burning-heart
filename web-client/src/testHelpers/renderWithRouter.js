import React from "react";
import { Router } from "react-router-dom";
import { render } from "react-testing-library";
import { createMemoryHistory } from "history";
import { Provider } from "react-redux";
import { store } from "../store";

const renderWithRouter = (
  ui,
  {
    route = "/",
    history = createMemoryHistory({ initialEntries: [route] })
  } = {}
) => {
  return {
    ...render(
      <Router history={history}>
        <Provider store={store}>{ui}</Provider>
      </Router>
    ),
    history
  };
};

export default renderWithRouter;
