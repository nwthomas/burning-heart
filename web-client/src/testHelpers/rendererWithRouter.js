import React from "react";
import { Router } from "react-router-dom";
import renderer from "react-test-renderer";
import { createMemoryHistory } from "history";
import { Provider } from "react-redux";
import { store } from "../store";

const rendererWithRouter = (
  ui,
  {
    route = "/",
    history = createMemoryHistory({ initialEntries: [route] })
  } = {}
) => {
  return {
    ...renderer.create(
      <Router history={history}>
        <Provider store={store}>{ui}</Provider>
      </Router>
    ),
    history
  };
};

export default rendererWithRouter;
