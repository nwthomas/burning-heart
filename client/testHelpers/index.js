import renderWithRouter from "./renderWithRouter";
import rendererWithRouter from "./rendererWithRouter";

export { renderWithRouter, rendererWithRouter };
