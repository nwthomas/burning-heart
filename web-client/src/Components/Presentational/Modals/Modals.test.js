import React from "react";
import "react-testing-library/cleanup-after-each";
import "jest-dom/extend-expect";
import { renderWithRouter, rendererWithRouter } from "../../../testHelpers";

import LoginModal from "./LoginModal";
import PaymentSendingModal from "./PaymentSendingModal";

describe("<LoginModal />", () => {
  it("should render the LoginModal component without crashing", () => {
    const push = jest.fn();
    const scrollToSpy = jest.fn();
    window.scrollTo = scrollToSpy;
    const helpers = renderWithRouter(<LoginModal />);
  });

  it("should match the snapshot of LoginModal", () => {
    const tree = rendererWithRouter(<LoginModal />);
    expect(tree.toJSON()).toMatchSnapshot();
  });
});

describe("<PaymentSendingModal />", () => {
  it("should match the snapshot of PaymentSendingModal", () => {
    const helpers = renderWithRouter(<PaymentSendingModal />);
  });

  it("should match the snapshot of PaymentSendingModal", () => {
    const tree = rendererWithRouter(<PaymentSendingModal />);
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
