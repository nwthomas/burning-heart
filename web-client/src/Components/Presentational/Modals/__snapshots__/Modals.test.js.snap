// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginModal /> should match the snapshot of LoginModal 1`] = `
<div
  className="modal__container"
>
  <div
    className="modal__box"
  >
    <h2
      className="modal__title"
    >
      
    </h2>
    <div
      className="modal__buttons modal__buttons--center"
    >
      <button
        className="modal__button"
        onClick={[Function]}
      >
        Okay
      </button>
    </div>
  </div>
</div>
`;

exports[`<PaymentSendingModal /> should match the snapshot of PaymentSendingModal 1`] = `
<div
  className="modal__container"
>
  <div
    className="modal__box"
  >
    <h2
      className="modal__title"
    >
      
    </h2>
    <div
      className="modal__buttons modal__buttons--center"
    >
      <button
        className="modal__button"
        onClick={[Function]}
      >
        Okay
      </button>
    </div>
  </div>
</div>
`;
